### PR TITLE
LIME-175 / 176 - DVLA | Field Validation, DVA | Field Validation

### PR DESCRIPTION
**Pull request by MMcC-GDS**
### Issue tracking

- [LIME-175](https://govukverify.atlassian.net/browse/LIME-175)
- [LIME-176](https://govukverify.atlassian.net/browse/LIME-176)

### CODE REVIEW

File: src/app/drivingLicence/fields.js [hunk 1]
```diff
@@ -72,22 +72,35 @@ module.exports = {
...
+      { type: "regexDrivingLicence", fn: (value) => value.match(/^(?=.{16}$)[A-Za-z]{1,5}9{0,4}[0-9](?:[05][1-9]|[16][0-2])(?:[0][1-9]|[12][0-9]|3[01])[0-9](?:99|[A-Za-z][A-Za-z9])(?![IOQYZioqyz01_])\w[A-Za-z]{2}$/) }
+    ],
+    classes: "govuk-input--width-10",
+  },
+  dvaLicenceNumber: {
+    type: "text",
+    journeyKey: "dvaLicenceNumber",
+    validate: [
+      "required",
+      { type: "regexDrivingLicence", fn: (value) => value.match(/^[0-9]{8}$/) }
```
Comment (ElliotMurphyGDS) [inline src/app/drivingLicence/fields.js:94]: we may need to make this field dependant on the licenceIssuerDependent field in the same way it was done for issue date/number fields

File: src/locales/en/fields.yml [hunk 2]
```diff
@@ -71,6 +81,9 @@ postcode:
   hint: "Enter the postcode in the address in section 8 of your licence"
   validation:
     default: "Enter the postcode as it appears on your driving licence"
+    maxlength: "Your postcode should be between 5 and 7 characters"
```
Comment (gds-jbentham) [inline src/locales/en/fields.yml:84]: Do we not need a min and max validation on postcode to trigger these

File: src/locales/en/fields.yml [hunk 3]
```diff
@@ -3,21 +3,31 @@ drivingLicenceNumber:
...
+    exactLength: "Your licence number should be 8 characters long"
+    numeric: "Your licence number should not include any symbols or spaces"
     limit: "Driving licence numbers do not start with 16 – enter the number as it appears on your driving licence"
 issueNumber:
   label: Issue Number
   hint: "This is the 2 digit number after the space in section 5 of your licence."
   validation:
-    default: "Enter the number as it appears on your driving licence"
+    default: "Enter the issue number as it appears on your driving licence"
+    exactLength: "Your issue number should be 2 numbers long"
```
Comment (gds-jbentham) [inline src/locales/en/fields.yml:22]: Do we not need a validation to trigger this

File: src/app/drivingLicence/fields.js [hunk 4]
```diff
@@ -72,22 +72,35 @@ module.exports = {
...
+      { type: "after", arguments: [new Date().toISOString().split("T")[0]] }
     ],
   },
   drivingLicenceNumber: {
     type: "text",
     journeyKey: "drivingLicenceNumber",
     validate: [
-      "required"
+      "required",
+      { type: "maxlength", arguments: [30] },
```
Comment (gds-jbentham) [inline src/app/drivingLicence/fields.js:83]: Should this not be 16

File: src/locales/en/fields.yml [hunk 5]
```diff
@@ -3,21 +3,31 @@ drivingLicenceNumber:
   hint: "This is the long number in section 5 on your licence for example HARRI559146MJ931 "
   validation:
     default: "Enter the number as it appears on your driving licence"
-    exactlength: "Your driving licence number should be 8-16 digits long"
-    numeric: "Your driving licence number should not include letters or symbols"
+    exactLength: "Your licence number should be 16 characters long"
+    numeric: "Your licence number should not include any symbols or spaces"
+    limit: "Driving licence numbers do not start with 16 – enter the number as it appears on your driving licence"
+dvaLicenceNumber:
+  label: Driving Licence number
```
Comment (ElliotMurphyGDS) [inline src/locales/en/fields.yml:10]: Only a small thing, can we make the L in licence a small l to be inline with the prototype

File: src/app/drivingLicence/fields.js (hunk 1 above)
Comment (MMcC-GDS) [reply src/app/drivingLicence/fields.js:95]: Added.

File: src/locales/en/fields.yml (hunk 2 above)
Comment (MMcC-GDS) [reply src/locales/en/fields.yml:84]: Added.

File: src/locales/en/fields.yml (hunk 3 above)
Comment (MMcC-GDS) [reply src/locales/en/fields.yml]: Added.

File: src/app/drivingLicence/fields.js (hunk 4 above)
Comment (MMcC-GDS) [reply src/app/drivingLicence/fields.js]: Corrected.

File: src/locales/en/fields.yml (hunk 5 above)
Comment (MMcC-GDS) [reply src/locales/en/fields.yml]: Pulled your changes.

**Review by ElliotMurphyGDS — APPROVED** [review]

**Pull request merged**